File: Programmers/Lv.1/체육복.py
def solution(n, lost, reserve):
    set_lost = set(lost) - set(reserve)
    set_reserve = set(reserve) - set(lost)
    
    for i in set_reserve:
        if i - 1 in set_lost:
            set_lost.remove(i - 1)
        elif i + 1 in set_lost:
            set_lost.remove(i + 1)
    return n - len(set_lost)

# def solution(n, lost, reserve):
#     answer = 0
#     for i in range(1, n+1):
#         if i not in lost: #안 잃어버린 학생
#             answer += 1
#         else:
#             if i in reserve: #잃어버렸지만 여분도 있는 학생
#                 answer += 1
#                 reserve.remove(i)
#                 lost.remove(i)

#     for i in lost: #잃어버리고 여분도 없어서 빌려야 하는 학생
#         if i-1 in reserve:
#             answer += 1
#             reserve.remove(i-1)

#         elif i+1 in reserve:
#             answer +=1
#             reserve.remove(i+1)

#     return answer

# def solution(n, lost, reserve):
#     _reserve = [r for r in reserve if r not in lost]
#     _lost = [l for l in lost if l not in reserve]

#     for r in _reserve:
#         f = r - 1
#         b = r + 1
#         if f in _lost:
#             _lost.remove(f)
#         elif b in _lost:
#             _lost.remove(b)

#     return n - len(_lost)

File: Programmers/Lv.1.Kotlin/체육복.kt
class Solution {
    fun solution(n: Int, lost: IntArray, reserve: IntArray): Int {
        var answer = 0
        var arr = Array(n + 1) { 1 }
        
        for (i in lost.indices) { arr[lost[i]]-- }
        for (i in reserve.indices) { arr[reserve[i]]++ }
        for (i in 1..n) {
            if (arr[i] == 0) {
                if (arr[i - 1] == 2) {
                    arr[i - 1]--
                    arr[i] = 1
                } else if (i + 1 <= n) {
                    if (arr[i + 1] == 2) {
                        arr[i + 1]--
                        arr[i] = 1
                    }
                }
            }
        }
        
        for (i in 1..n) {
            if (arr[i] > 0) answer++
        }
        
        return answer
    }

    // fun solution(n: Int, lost: IntArray, reserve: IntArray): Int {
    //     var answer = n
    //     var lostSet = lost.toSet() - reserve.toSet()
    //     var reserveSet = (reserve.toSet() - lost.toSet()) as MutableSet

    //     for (i in lostSet) {
    //         when {
    //             i + 1 in reserveSet -> reserveSet.remove(i + 1)
    //             i - 1 in reserveSet -> reserveSet.remove(i - 1)
    //             else -> answer--
    //         }
    //     }
    //     return answer
    // }
}